parser/gotest/internal/collector/collector.go

```diff
@@ -4,13 +4,24 @@ package collector
 
 import (
 	"sort"
+	"strings"
 	"time"
+	"unicode"
 )
 
 // line is a single line of output captured at some point in time.
 type line struct {
 	Timestamp time.Time
-	Text      string
+	text      string
+}
+
+func (l line) SafeText() string {
+	return strings.Map(func(r rune) rune {
+		if unicode.IsControl(r) && !unicode.IsSpace(r) {
```

Comment:
Some characters considered whitespace by `unicode.IsSpace` are illegal in the [XML spec](https://www.w3.org/TR/xml/#charsets) (for example `\v` and `\f`). And `unicode.IsControl` doesn't exclude other illegal character ranges either.

Let's use the exact character range defined by the spec in this function instead.

Reply:
Great point. Unfortunately `xml.Escpe` does not handle whitespaces as I'd expect and not translate `\t` into spaces